File: go/app/external_token.go
package app

import (
	"net/http"
	"os"
)

func (app *App) externalToken(w http.ResponseWriter, r *http.Request) {
	data := map[string]interface{}{
		"title": "Using external API for token based authentication",
		"env": map[string]interface{}{
			"NT_ALIAS_DEMO":           os.Getenv("NT_ALIAS_DEMO"),
			"NT_ALIAS_DEFAULT":        "demo",
			"NT_TOKEN_PUBLIC_KEY_URL": os.Getenv("NT_TOKEN_PUBLIC_KEY_URL"),
		},
	}

	app.render(w, "external_token", data)
}

File: php/routes/ExternalToken.php
<?php
/*
This file is part of the Nervatura Framework
http://nervatura.com
Copyright © 2011-2023, Csaba Kappel
License: LGPLv3
https://raw.githubusercontent.com/nervatura/nervatura/master/LICENSE
*/

class ExternalToken {
  function init($f3){
    $f3->set('title', 'Using external API for token based authentication');
    $f3->set('env', array(
      'NT_TOKEN_PUBLIC_KEY_URL'=>getenv('NT_TOKEN_PUBLIC_KEY_URL'),
      'NT_ALIAS_DEFAULT'=>getenv('NT_ALIAS_DEFAULT'),
      'NT_ALIAS_DEMO'=>getenv('NT_ALIAS_DEMO')
    ));
  }
  
  public function get($f3){
    $this->init($f3);
    $this->sendResult($f3);
  }

  function sendResult($f3){
    $f3->set('content', 'external_token.html');
    echo Template::instance()->render('layout.html');
  }
}

?>